core/src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala

```diff
@@ -119,10 +119,23 @@ case class SubmissionStatusResponse(
   methodConfigurationName: String,
   submissionEntity: AttributeEntityReference,
   workflows: Seq[Workflow],
+  workflowCostMap: Map[String, Float] = Map.empty,
   status: SubmissionStatus,
   workflowFailureMode: Option[WorkflowFailureMode] = None
 ) {
-  def this(submission: Submission, rawlsUser: RawlsUser) = this(submission.submissionId, submission.submissionDate, rawlsUser.userEmail.value, submission.methodConfigurationNamespace, submission.methodConfigurationName, submission.submissionEntity, submission.workflows, submission.status, submission.workflowFailureMode)
+  def this(submission: Submission, workflowCostMap: Map[String, Float], rawlsUser: RawlsUser) =
+    this(
+      submissionId = submission.submissionId,
+      submissionDate = submission.submissionDate,
+      submitter = rawlsUser.userEmail.value,
+      methodConfigurationNamespace = submission.methodConfigurationNamespace,
+      methodConfigurationName = submission.methodConfigurationName,
+      submissionEntity = submission.submissionEntity,
+      workflows = submission.workflows,
+      workflowCostMap = workflowCostMap,
+      status = submission.status,
+      workflowFailureMode = submission.workflowFailureMode
+    )
```

Comment:
I'm pondering this. Instead of having `workflowCostMap` be an additional field on `SubmissionStatusResponse`, what if we added a `cost` field to `Workflow` as an `Option[Float] = None`? `Workflow` is used in a lot of places and so this change could have wide impact, but it feels like a better encapsulation.

Either way, this will need an update to swagger to denote the new fields.

Reply:
I think that I agree with you on this. I guess the big question would be when to hit BigQuery.

Reply:
oh, I don't think you have to change too much about that. the flow would be the same, and you'd still pass `workflowCostMap` as an argument when creating `SubmissionStatusResponse` ... but instead of `SubmissionStatusResponse` having a dedicated field, the costs in `workflowCostMap` would be annotated onto the workflows. That make sense?

Reply:
I think you'd end up with something like this (unproven, uncompiled):
```
  def getSubmissionStatus(workspaceName: WorkspaceName, submissionId: String): Future[PerRequestMessage] = {
    val submissionWithoutCosts = dataSource.inTransaction { dataAccess =>
      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
        withSubmission(workspaceContext, submissionId, dataAccess) { submission =>
          withUser(submission.submitter, dataAccess) { user =>
            DBIO.successful(submission, user)
          }
        }
      }
    }

    submissionWithoutCosts flatMap {
      case (submission, user) => {
        val allWorkflowIds: Seq[String] = submission.workflows.flatMap(_.workflowId)
        submissionCostService.getWorkflowCosts(workspaceName.namespace, allWorkflowIds, GoogleProject(workspaceName.namespace)) map { costMap =>
          val costedWorkflows = submission.workflows.map { wf =>
            wf.workflowId match {
              case Some(wfid) => wf.copy(cost = costMap.get(wfid)) // this will only compile once cost is added as a field on Workflow
              case None => wf
            }
          }
          val costedSubmission = submission.copy(workflows = costedWorkflows)
          RequestComplete((StatusCodes.OK, new SubmissionStatusResponse(costedSubmission, costMap, user)))
        }
      }
    }
  }
```

Reply:
and after sleeping on this, I suggest we add `cost` to each workflow, *and* add a `cost` to the submission itself that has `costMap.values.sum`. It's technically duplicate data (callers of the API could do the sum themselves) but it feels like a nice convenience.

Reply:
Great point!